gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/process/repository/synchronizer/api/PlanAppender.java

```diff
@@ -71,11 +73,12 @@ public List<ApiReactorDeployable> appends(final List<ApiReactorDeployable> deplo
             })
             .filter(deployable -> {
                 ReactableApi<?> reactableApi = deployable.reactableApi();
-                if (reactableApi.getDefinition() instanceof io.gravitee.definition.model.v4.Api) {
-                    io.gravitee.definition.model.v4.Api v4definition = (io.gravitee.definition.model.v4.Api) reactableApi.getDefinition();
-                    return v4definition.getPlans() != null && !v4definition.getPlans().isEmpty();
-                } else if (reactableApi.getDefinition() instanceof io.gravitee.definition.model.Api) {
-                    return !((io.gravitee.definition.model.Api) reactableApi.getDefinition()).getPlans().isEmpty();
+                if (reactableApi.getDefinition() instanceof io.gravitee.definition.model.v4.Api v4Api) {
+                    return v4Api.getPlans() != null && !v4Api.getPlans().isEmpty();
+                } else if (reactableApi.getDefinition() instanceof NativeApi nativeApi) {
+                    return nativeApi.getPlans() != null && !nativeApi.getPlans().isEmpty();
+                } else if (reactableApi.getDefinition() instanceof io.gravitee.definition.model.Api api) {
+                    return !api.getPlans().isEmpty();
```

Comment:
`return api.getPlans() != null && !api.getPlans().isEmpty();` ?

We can also add an info log saying that the API is excluded because no plan is associated? (it is currently silently excluded)

Reply:
I keep the same condition as it was in the code but I can align yes.
and yes I will add a log

Reply:
I can add the check but in V2 the getPlans is returning an empty list